templates.rst

```diff
@@ -0,0 +1,1061 @@
+  these comments are not included in the rendered page).
+
+You can't run PHP code inside Twig templates, but Twig provides utilities to
+run some logic in the templates. For example, **filters** modify contents before
+being rendered, like the ``upper`` filter to uppercase contents:
+
+.. code-block:: twig
+
+    {{ title|upper }}
+
+Twig comes with a long list of `tags`_, `filters`_ and `functions`_ that are
+available by default. In Symfony applications you can also use these
+:doc:`Twig filters and functions defined by Symfony </reference/twig_reference>`
+and you can :doc:`create your own Twig filters and functions </templating/twig_extension>`.
+
+Twig is fast in the ``prod`` :doc:`environment <configuration-environments>`
+(because templates are compiled into PHP and cached automatically), but
+convenient to use in the ``dev`` environment (because templates are recompiled
+automatically when you change them).
+
+Twig Configuration
+~~~~~~~~~~~~~~~~~~
+
+Twig defines several configuration options to define things like the format used
+to display numbers and dates, the template caching, etc. Read the
+:doc:`Twig configuration reference </reference/configuration/twig>` to learn about them.
+
+Creating Templates
+------------------
+
+Template Naming
+~~~~~~~~~~~~~~~
+
+Symfony recommends the following for template names:
+
+* Use `snake case`_ for filenames and directories (e.g. ``blog_posts.twig``,
+  ``admin/default_theme/blog/index.twig``, etc.);
+* Define two extensions for filenames (e.g. ``index.html.twig`` or
+  ``blog_posts.xml.twig``) being the first extension (``html``, ``xml``, etc.)
+  the final format that the template will generate.
+
+Although templates usually generate HTML contents, they can generate any
+text-based format. That's why the two-extension convention simplifies the way
+templates are created and rendered for multiple formats.
+
+For example, if the route includes the :ref:`special _format routing parameter <routing-format-parameter>`
+(e.g. ``/index.{_format}``) you can browse different URLs (``/index.html``,
+``/index.xml``, ``/index.json``) to render different templates.
+
+Although Symfony doesn't select the template automatically based on the request
+format, you can do that yourself. Instead of rendering the ``'index.html.twig'``
+template, if your application supports different formats, you can render
+``'index.'.$format.'.twig'``, where ``$format = $request->getRequestFormat()``.
+
+Template Location
+~~~~~~~~~~~~~~~~~
+
+Templates are stored by default in the ``templates/`` directory. When a service
+or controller renders the ``product/index.html.twig`` template, they are actually
+referring to the ``<your-project>/templates/product/index.html.twig`` file.
+
+The default templates directory is configurable with the
+:ref:`twig.default_path <config-twig-default-path>` option and you can add more
+template directories :ref:`as explained later <templates-namespaces>` in this article.
+
+Template Variables
```

Comment:
If we start this section with a nice, real-world example, then this can become (or at least continue) that example. For... example, if the initial example shows passing in a simple string variable and printing it, this section could be called `Accessing Object Data in Twig` and it would pass a new `comment` variable from the controller into Twig, and then show how to access its data.